internal/envoy/v3/cluster.go

```diff
@@ -312,3 +331,17 @@ func protocolOptions(explicitHTTPVersion HTTPVersionType, idleConnectionTimeout
 		"envoy.extensions.upstreams.http.v3.HttpProtocolOptions": protobuf.MustMarshalAny(&options),
 	}
 }
+
+// slowStartConfig returns the slow start configuration.
+func slowStartConfig(slowStartConfig *dag.SlowStartConfig) *envoy_cluster_v3.Cluster_SlowStartConfig {
+	return &envoy_cluster_v3.Cluster_SlowStartConfig{
+		SlowStartWindow: protobuf.Duration(slowStartConfig.Window),
+		Aggression: &envoy_core_v3.RuntimeDouble{
+			DefaultValue: slowStartConfig.Aggression,
+			RuntimeKey:   "contour.slowstart.aggression",
```

Comment:
interesting this is also configurable via RTDS, not sure if we will need to take advantage of that though

Reply:
True, I did not see how to take advantage of it either, but it seems that (almost) all `Runtime*` message types have validation rules in `.proto` which makes the `RuntimeKeys` required.  For example in this case [`RuntimeDouble.runtime_key`](https://github.com/envoyproxy/envoy/blob/8f2674fa6a024bf0eee502834b93e6e82f957bd1/api/envoy/config/core/v3/base.proto#L287) has validation rule:

```
string runtime_key = 2 [(validate.rules).string = {min_len: 1}];
```

that causes config to be rejected if runtime key is not given:

```
[2022-10-06 06:03:23.608][1][warning][config] [source/common/config/grpc_subscription_impl.cc:126] 
gRPC config for type.googleapis.com/envoy.config.cluster.v3.Cluster rejected:
Proto constraint validation failed
(ClusterValidationError.RoundRobinLbConfig: embedded message failed validation | 
caused by RoundRobinLbConfigValidationError.SlowStartConfig: embedded message failed validation |
caused by SlowStartConfigValidationError.Aggression: embedded message failed validation |
caused by RuntimeDoubleValidationError.RuntimeKey: value length must be at least 1 characters)
```

So I have assumed these are mandatory, even if we would never use them.

Curiously, there are at least some cases where it is not required, like when we configure ratelimit [here](https://github.com/projectcontour/contour/blob/d42b7f419050a33625adf691e9321c543acedf5f/internal/envoy/v3/ratelimit.go#L49-L54). In this case the message definition is [missing the validation rule](https://github.com/envoyproxy/envoy/blob/8f2674fa6a024bf0eee502834b93e6e82f957bd1/api/envoy/config/core/v3/base.proto#L486-L495). Maybe that is just by mistake, since that is the only `runtime_key` that is missing validation in [`base.proto`](https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/core/v3/base.proto).